test/unit/foreman/http_proxy_test.rb

```diff
@@ -88,22 +88,20 @@ class DummyHTTPAdapter
 
     setup do
       excon_connection.stubs(:http_proxy).returns(http_proxy)
-      excon_connection.stubs(:orig_request).returns(true)
       excon_connection.stubs(:setup_proxy).returns
       excon_connection.stubs(:proxy_http_request?).returns(true)
     end
 
     test 'set @data[:proxy] to proxy' do
-      excon_connection.request({})
-      assert_equal http_proxy,
-                   excon_connection.instance_variable_get(:@data)[:proxy]
+      stub_request(:get, "http://#{request_host}/features").to_return(status: 200, body: "", headers: {})
+      excon_connection.request_with_foreman_extensions({host: request_host, path: "/features", headers: {}})
```

Comment:
Let me explain, on this line I need to call the foreman extended method, unless it has an explicit name, I can't do that because webmock always shadows original and foreman versions and I can't access it. Maybe via some dirty trick like `object.successors.second.call(:request)`?